prisma-fmt/src/code_actions/relations.rs

```diff
@@ -221,3 +223,98 @@ fn create_missing_unique<'a>(
 
     TextEdit { range, new_text }
 }
+
+/// For schema's with emulated relations,
+/// If the referenced side of the relation does not point to a unique
+/// constraint, the action adds the attribute.
+///
+/// If referencing a single field:
+///
+/// ```ignore
+/// model A {
+///     id      Int @id
+///     field1  B   @relation(fields: [bId], references: [id])
+///                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ // Warn
+///     bId     Int
+///
+///     // <- suggest @@index([bId]) here
+/// }
+///
+/// model B {
+///     id Int @id
+///     as A[]
+/// }
+/// ```
+///
+/// If referencing multiple fields:
+///
+/// ```ignore
+/// model A {
+///     id      Int @id
+///     field1  B   @relation(fields: [bId1, bId2, bId3], references: [id1, id2, id3])
+///                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ // Warn
+///     bId1    Int
+///     bId2    Int
+///     bId3    Int
+///
+///     // <- suggest @@index([bId1, bId2, bId3]) here
+/// }
+///
+/// model B {
+///     id1 Int
+///     id2 Int
+///     id3 Int
+///     as  A[]
+///
+///     @@id([id1, id2, id3])
+/// }
+/// ```
+pub(super) fn add_reference_index(
+    actions: &mut Vec<CodeActionOrCommand>,
+    params: &CodeActionParams,
+    schema: &str,
+    relation: RelationFieldWalker<'_>,
+) {
+    let Some(fields) = relation.fields() else { return; };
+    if relation.model().indexes().any(|index| {
+        index
+            .fields()
+            .zip(fields.clone())
+            .all(|(index_field, relation_field)| index_field.field_id() == relation_field.field_id())
+    }) {
+        return;
+    }
+
+    let model_end = Range {
+        start: offset_to_position(relation.model().ast_model().span().end, schema).unwrap(),
+        end: offset_to_position(relation.model().ast_model().span().end, schema).unwrap(),
+    };
+
+    let text = TextEdit {
+        range: model_end,
+        new_text: String::from("new_text"),
+    };
+    let mut changes = HashMap::new();
+    changes.insert(params.text_document.uri.clone(), vec![text]);
+
+    let edit = WorkspaceEdit {
+        changes: Some(changes),
+        ..Default::default()
+    };
+
+    let diagnostics = super::diagnostics_for_span(
+        schema,
+        &params.context.diagnostics,
+        relation.relation_attribute().unwrap().span(),
+    );
+
+    let action = CodeAction {
+        title: String::from("Index relations"),
```

Comment:
```suggestion
        title: String::from("Add an index for the relation fields"),
```
Maybe?

Reply:
Thoughts on? ::
```suggestion
        title: String::from("Add relation fields as indexes"),
```

Reply:
`indexes` would mean more than one index to me. And what we do here would be to add only one index on the field(s)

Reply:
I do see what you mean, it's a weird spot in English. I was coming at it from the perspective of: it's an array of indices `@@index([indices])`, like each field getting added is in and of itself an index

Reply:
So when a "single index structure holds references to multiple fields", we can also call it a compound index.
Here we would need to either 
- have 1 text that describes adding an index (no matter the number of fields)
- have 2 text
	- when `@relation` has only one field show, for example: "Add an index for the relation field"
	- when `@relation` has more than one field show, for example: "Add a compound index for the relation fields"

Reply:
> it's an array of indices `@@index([indices])`, like each field getting added is in and of itself an index

That is not the case here. The full thing is the index, and the array is the list of fields that the index covers.

Reply:
Fair enough!